src/Console/GenerateCommandBuilder.php

```diff
@@ -0,0 +1,180 @@
+                    return Command::FAILURE;
+                }
+                $tempDirPath = $this->getSvgTempPath();
+                $this->ensureDirExists($tempDirPath);
+
+                $output->writeln("Discovering source SVGs for icon sets...");
+                foreach ($this->iconSets as $iconSetConfig) {
+                    /**
+                     * @var IconSetConfig $iconSetConfig
+                     */
+                    $iconSetConfig->setSourcePath($this->getSvgSourcePath())
+                                ->setTempPath($this->getSvgTempPath())
+                                ->setDestinationPath($this->getSvgDestinationPath());
+                    $iconSetName = $iconSetConfig->name;
+                    $output->writeln("Processing '{$iconSetName}' icon set SVGs.");
+                    // Setup build dir for type
+                    $iconSetTmpDir = $tempDirPath . DIRECTORY_SEPARATOR . $iconSetName;
+                    $this->ensureDirExists($iconSetTmpDir);
+
+                    $iconFileList = $this->getDirectoryFileList($this->getSvgSourcePath() . DIRECTORY_SEPARATOR . $iconSetName, $iconSetConfig);
+                    $this->updateSvgs($iconSetConfig, $iconFileList);
+                    $output->writeln("Completed processing for '{$iconSetName}' svgs.");
+                }
+
+                $output->writeln("Cleaning up the build directory...");
+                $this->deleteDirectory(static::getSvgTempPath());
+                $output->writeln("Done!");
+
+                return Command::SUCCESS;
+            })
+            ->run();
+    }
+
+    private function getSvgSourcePath(): string
+    {
+        if ($this->npmPackageName !== null) {
+            return $this->baseDirectory . DIRECTORY_SEPARATOR .
+                    'node_modules' . DIRECTORY_SEPARATOR .
+                    $this->npmPackageName . DIRECTORY_SEPARATOR .
+                    ltrim($this->sourceDirectory, DIRECTORY_SEPARATOR);
+        }
+
+        return $this->baseDirectory . DIRECTORY_SEPARATOR . ltrim($this->sourceDirectory, DIRECTORY_SEPARATOR);
+    }
+
+    private function getSvgTempPath(): string
+    {
+        return $this->baseDirectory . DIRECTORY_SEPARATOR . 'build';
+    }
+
+    private function getSvgDestinationPath(): string
+    {
+        return $this->baseDirectory . DIRECTORY_SEPARATOR . 'resources' . DIRECTORY_SEPARATOR . 'svg';
+    }
+
+    private function ensureDirExists($dirPath)
+    {
+        if (!is_dir($dirPath)) {
+            if (!mkdir($dirPath) && !is_dir($dirPath)) {
+                throw new \RuntimeException(sprintf('Directory "%s" was not created', $dirPath));
+            }
+        }
+    }
+
+    /**
+     * @param string $dirPath
+     *
+     * @return array<array<string>>
+     */
+    private function getDirectoryFileList(string $dirPath, IconSetConfig $iconSetConfig): array
+    {
+        return array_diff(scandir($dirPath), ['..', '.']);
```

Comment:
For Icon packs with the "main" set in a root dir, the alternative sets would be in directories.
So to support those, we will need to expand this to remove all directories along with `.` and `..`.

Reply:
I think by using Laravel's `Filesystem::files()` we get the benefit of not worrying about this potential issue. So at the very least those icon sets won't be broken now.